.github/scripts/ci_tests.sh

```diff
@@ -56,44 +66,119 @@ for arg in $RUN_BOARDS_BUILD
 do
     case "$arg" in
         "-esp")
-            echo -e "::group::Building sample for ESPRESSIF ESP32 port"
-            sample_build "ESPRESSIF" "esp32" "build"
+            echo -e "::group::Building sample for ESPRESSIF ESP32 port - Debug"
+            sample_build "ESPRESSIF" "esp32" "build" "Debug"
+            exit_if_binary_does_not_exist "./demos/projects/ESPRESSIF/esp32/build" "azure_iot_freertos_esp32.bin"
+            sample_build "ESPRESSIF" "aziotkit" "build" "Debug"
+            exit_if_binary_does_not_exist "./demos/projects/ESPRESSIF/aziotkit/build" "azure_iot_freertos_esp32.bin"
+
+            rm -rf build
+
+            echo -e "::group::Building sample for ESPRESSIF ESP32 port - Release"
+            sample_build "ESPRESSIF" "esp32" "build" "Release"
             exit_if_binary_does_not_exist "./demos/projects/ESPRESSIF/esp32/build" "azure_iot_freertos_esp32.bin"
-            sample_build "ESPRESSIF" "aziotkit" "build"
+            sample_build "ESPRESSIF" "aziotkit" "build" "Release"
             exit_if_binary_does_not_exist "./demos/projects/ESPRESSIF/aziotkit/build" "azure_iot_freertos_esp32.bin"
             ;;
         "-nxp")
             fetch_freertos
 
-            echo -e "::group::Building sample for NXP mimxrt1060 port"
-            sample_build "NXP" "mimxrt1060" "build_nxp_mimxrt1060"
+            echo -e "::group::Building sample for NXP mimxrt1060 port - Debug"
+            sample_build "NXP" "mimxrt1060" "build_nxp_mimxrt1060" "Debug"
             exit_if_binary_does_not_exist "build_nxp_mimxrt1060" "iot-middleware-sample.elf"
             exit_if_binary_does_not_exist "build_nxp_mimxrt1060" "iot-middleware-sample-pnp.elf"
+
+            echo -e "::group::Print Size for Debug"
+            grep -A9 "Linking C executable demos/projects/NXP/mimxrt1060/iot-middleware-sample.elf" build.txt
```

Comment:
Nit - since you're intending this as a build output helper only the grep can work, though consider greping on "memory region" which seems less likely to change than the "linking..." always being before size in future linker output changes.

Reply:
Yea I hear that. If the linker output changes, this grep should then fail and the build will fail so we can likely resolve it then.